rskj-core/src/main/java/co/rsk/peg/PegUtilsLegacy.java

```diff
@@ -0,0 +1,393 @@
+                try {
+                    // Consider transactions that have an input with a redeem script of type P2SH ERP FED
+                    // to be "future transactions" that should not be pegins. These are gonna be considered pegouts.
+                    // This is only for backwards compatibility reasons. As soon as RSKIP353 activates,
+                    // pegins to the new federation should be valid again.
+                    // There's no reason for someone to send an actual pegin of this type before the new fed is active.
+                    // TODO: Remove this if block after RSKIP353 activation
+                    if (!activations.isActive(ConsensusRule.RSKIP353) &&
+                            (redeemScriptParser.getMultiSigType() == RedeemScriptParser.MultiSigType.P2SH_ERP_FED ||
+                                 redeemScriptParser.getMultiSigType() == RedeemScriptParser.MultiSigType.FAST_BRIDGE_P2SH_ERP_FED)) {
+                        String message = "Tried to register a transaction with a P2SH ERP federation redeem script before RSKIP353 activation";
+                        logger.warn("[isValidPegInTx] {}", message);
+                        throw new ScriptException(message);
+                    }
+                    Script inputStandardRedeemScript = redeemScriptParser.extractStandardRedeemScript();
+                    if (activeFederations.stream().anyMatch(federation -> federation.getStandardRedeemScript().equals(inputStandardRedeemScript))) {
+                        return false;
+                    }
+
+                    Script outputScript = ScriptBuilder.createP2SHOutputScript(inputStandardRedeemScript);
+                    if (outputScript.equals(retiredFederationP2SHScript)) {
+                        return false;
+                    }
+                } catch (ScriptException e) {
+                    // There is no redeem script, could be a peg-in from a P2PKH address
+                }
+            }
+        }
+
+        Wallet federationsWallet = new BridgeBtcWallet(btcContext, activeFederations);
+        Coin valueSentToMe = tx.getValueSentToMe(federationsWallet);
+
+        boolean isUTXOsOrTxAmountBelowMinimum =
+            activations.isActive(RSKIP293) ? isAnyUTXOAmountBelowMinimum(
+                minimumPegInTxValue,
+                tx,
+                federationsWallet
+            ) : valueSentToMe.isLessThan(minimumPegInTxValue); // Legacy minimum validation against the total amount
+
+        if (!isUTXOsOrTxAmountBelowMinimum) {
+            return true;
+        }
+
+        logger.warn(
+            activations.isActive(RSKIP293)?
+                "[btctx:{}] Someone sent to the federation UTXOs amount less than {} satoshis":
+                "[btctx:{}] Someone sent to the federation less than {} satoshis",
+            tx.getHash(),
+            minimumPegInTxValue
+        );
+        return false;
```

Comment:
```suggestion
        if (isUTXOsOrTxAmountBelowMinimum) {
          logger.warn(
              activations.isActive(RSKIP293)?
                  "[btctx:{}] Someone sent to the federation UTXOs amount less than {} satoshis":
                  "[btctx:{}] Someone sent to the federation less than {} satoshis",
              tx.getHash(),
              minimumPegInTxValue
          );
          return false;
        }
        return true;
```
Lets Avoid Negations - ! In If Statements, Makes Code Hard To Read.